src/Workspaces.cc

```diff
@@ -399,7 +399,23 @@ Workspaces::insert(PWinObj *wo, bool raise)
         }
     }
 
-    _wobjs.insert(it, wo);
+    it = _wobjs.insert(it, wo);
+
+    // Iconify fullscreen windows that may hide the newly raised
+    // window (while it likely still gets focus)
+    //
+    // Note that if FullscreenDetect is true (default) we should be
+    // able to handle fully maximized windows here too, so we don't
+    // need extra checks.
+    if (raise && iconify_fullscreen && wo->getLayer() != LAYER_DOCK) {
+        for (++it; it != _wobjs.end(); ++it) {
+            if ((*it)->getWorkspace() == wo->getWorkspace()
```

Comment:
Is that check even required, if it's mapped it should be in the way right? (Considering sticky windows as well)

Reply:
Yeah I was nervous and tried to keep the checks as tight as possible. But hmm.. how does it work i multi heads? I suppose all the screens still belong to one workspace (i.e. still safe to remove this check)?

Reply:
The workspaces span all available heads yes, and a window can occupy multiple heads. Maybe it would be better to do:

```
(*it)->isMapped()
&& (*it)->isFullscreen()
&& wo->isCoveredBy(*it)
```

The isCoveredBy doesn't exist but that way no fullscreen windows get iconified if they aren't actually in the way?

Reply:
This way isCoveredBy has the geometry of both the to-be-active and currently-active windows too, so in theory it would work even without isFullscreen()? Which extends the scope of this PR by a bit: if any window is completely hidden by another one on a higher layer, that other window gets "something" (most likely iconfied, but I suppose shade is an option for non-fullscreen, decorated windows too).